components/raftstore-v2/src/raft/apply.rs

```diff
@@ -1,23 +1,108 @@
 // Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
 
 use engine_traits::{KvEngine, RaftEngine};
+use kvproto::{raft_cmdpb::RaftCmdResponse, raft_serverpb::RegionLocalState};
+use raftstore::store::fsm::apply::DEFAULT_APPLY_WB_SIZE;
 use slog::Logger;
 
 use super::Peer;
-use crate::tablet::CachedTablet;
+use crate::{
+    fsm::ApplyResReporter,
+    router::{ApplyRes, CmdResChannel},
+    tablet::CachedTablet,
+};
 
 /// Apply applies all the committed commands to kv db.
-pub struct Apply<EK: KvEngine> {
-    tablet: CachedTablet<EK>,
-    logger: Logger,
+pub struct Apply<EK: KvEngine, R> {
+    remote_tablet: CachedTablet<EK>,
+    tablet: EK,
```

Comment:
Why two fields? Isn't `remote_tablet.cached() == tablet`?

Reply:
`remote_tablet` is only used for updating local tablet. It's not safe to query it directly as peer may change the tablet when applying a snapshot.